src/batched/KokkosBatched_SetIdentity_Decl.hpp

```diff
@@ -34,6 +34,28 @@ namespace KokkosBatched {
              const AViewType &A);
     };
 
+
+    ///
+    /// Selective Interface
+    ///
+    template<typename MemberType,
+             typename ArgMode>
+    struct SetIdentity {
+      template<typename AViewType>
+      KOKKOS_FORCEINLINE_FUNCTION
+      static int
+      invoke(const MemberType &member,
+             const AViewType &A) {
+        int r_val = 0;
+        if (std::is_same<ArgMode,Mode::Serial>::value) {
```

Comment:
See above.  A tag-based design could let you just write two overloads, instead of an if statement.

Reply:
It is all compile time if statement. So, I thought it should not affect the behavior nor performance. However, what you suggested is probably a more clean way to write a code.